lib/record_store/provider/oracle_cloud_dns.rb

```diff
@@ -0,0 +1,195 @@
+require 'oci'
+
+module RecordStore
+  class Provider::OracleCloudDNS < Provider
+    class Error < StandardError; end
+
+    class << self
+      def client
+        config = OCI::Config.new
+        config.user = secrets['user']
+        config.fingerprint = secrets['fingerprint']
+        config.key_content = secrets['key_content']
+        config.tenancy = secrets['tenancy']
+        config.region = secrets['region']
+        config.validate
+        OCI::Dns::DnsClient.new(config: config)
+      end
+
+      # Downloads all the records from the provider.
+      #
+      # Returns: an array of `Record` for each record in the provider's zone
+      def retrieve_current_records(zone:, stdout: $stdout)
+        client.get_zone_records(zone)
+          .flat_map { |response| response.data.items }
+          .flat_map { |api_record| build_from_api(api_record) }
+          .compact
+      rescue StandardError
+        stdout.puts "Cannot build recrod for zone: #{zone}"
```

Comment:
```suggestion
        stdout.puts "Cannot build record for zone: #{zone}"
```

Reply:
Note also that this silence the error as instead of having it propagate, we're just "printing" that there was an error, but then continue as if it never happened. Is that what we want ?

Reply:
`def retrieve_current_records(zone:, stdout: $stdout)`
DNSimple and Google Cloud DNS are using stdout like this way.
I took away from them to use it instead of putting ignore Rubocop comment.
[Here's reference](https://github.com/Shopify/record_store/blob/master/lib/record_store/provider/dnsimple.rb#L15)

Reply:
Yes, but as you can see, they propagate the exception: https://github.com/Shopify/record_store/blob/master/lib/record_store/provider/dnsimple.rb#L17 (from your reference)